src/solvers/highs.rs

```diff
@@ -116,30 +116,53 @@ impl HighsParallelType {
     }
 }
 
-#[derive(Debug, Clone, Copy)]
-struct HighsOptions {
-    presolve: HighsPresolveType,
-    solver: HighsSolverType,
-    parallel: HighsParallelType,
-    mip_abs_gap: Option<f32>,
-    mip_rel_gap: Option<f32>,
-    time_limit: f64,
-    threads: u32,
+/// A HiGHS option value.
+#[derive(Debug, Clone)]
+pub enum HighsOptionValue {
+    /// String option
+    String(String),
+    /// Boolean option
+    Bool(bool),
+    /// Integer option
+    Int(i32),
+    /// Floating point number option
+    Float(f64),
 }
-
-impl Default for HighsOptions {
-    fn default() -> Self {
-        Self {
-            presolve: HighsPresolveType::Choose,
-            solver: HighsSolverType::Choose,
-            parallel: HighsParallelType::Choose,
-            mip_abs_gap: None,
-            mip_rel_gap: None,
-            time_limit: f64::MAX,
-            threads: 0,
+impl HighsOptionValue {
```

Comment:
Shouldn't we use the existing HighsOptionValue from the highs crate instead ? https://docs.rs/crate/highs/latest/source/src/options.rs

Reply:
I didn’t find how, as this is a trait:
* is not public
* is not object safe, as it takes `self` as a parameter (and not a reference), so I can’t save some `Box<HighsOptionValue>`

Reply:
true